docs/proposal/20210823-gpu-support.md

```diff
@@ -0,0 +1,164 @@
+# GPU support CAPV
+
+```text
+---
+title: GPU support CAPV
+authors:
+  - "@geetikabatra"
+reviewers:
+  - "@vijaykumar"
+  - "@ankitaswamy"
+  - "@ashutoshkumar"
+  
+creation-date: 2021-08-23
+last-updated: 2021-08-25
+status: WIP
+```
+
+## Table of Contents
+
+* [Single Controller Multitenancy](#gpu-support)
+  * [Glossary](#glossary)
+  * [Summary](#summary)
+  * [Motivation](#motivation)
+    * [Goals](#goals)
+  * [Proposal](#proposal)
+    * [User Stories](#user-stories)
+      * [Story 1 - using vGPU](#story-1---using-vgpu)
+      * [Story 2 - GPU Direct implementation](#story-2---gpu-direct-implementation)
+      * [Story 3 - PCI passthrough for single node Customer](#story-3---pci-passthrough-for-single-node-customer)
+    * [Requirements](#requirements)
+      * [Functional Requirements](#functional-requirements)
+      * [Non-Functional Requirements](#non-functional-requirements)
+    * [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+      * [Controller Changes](#controller-changes)
+  * [Additional Details](#additional-details)
+    * [Test Plan](#test-plan)
+    * [Graduation Criteria](#graduation-criteria)
+  * [Implementation History](#implementation-history)
+
+## Glossary
+
+* CAPV - An abbreviation of Cluster API Provider vSphere
+
+## Summary
+
+CAPV currently does not support GPU-based infrastructure.
+
+This proposal outlines ways to provision GPU support in CAPV. The proposed changes will maintain backward compatibility and maintain the existing behavior without any extra user configuration.
+
```

Comment:
n_p:
To make these lines more clear, I was thinking to put some lines explaining what does GPU support in CAPV means?

e.g. 

```
CAPV currently does not support provisioning of GPU-based worker nodes and hence limits the deployment of workloads that needs GPU 

This proposal outlines ways to enable GPU support in CAPV. This means that the k8s cluster created via CAPV can have GPU enabled nodes that can be finally consumed by the workloads(pods) that request GPU.   The proposed changes will maintain backward compatibility and maintain the existing behaviour without any extra user configuration.

"As an end user, I want to deploy GPU accelerated workloads on the Kubernetes cluster created via CAPV

```

Reply:
@sonasingh46. I think it is self-explanatory.  But, if you strongly feel I should rephrase it. I would certainly do that.

Reply:
I was only confused by `provision GPU support in CAPV`. It is just a nit pick from me. Let us see what others have to say. Should be fine.